types/aws-lambda/trigger/dynamodb-stream.d.ts

```diff
@@ -3,19 +3,27 @@ import { Handler } from "../handler";
 // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
 export type DynamoDBStreamHandler = Handler<DynamoDBStreamEvent, DynamoDBBatchResponse | void>;
 
+// eslint-disable-next-line @definitelytyped/strict-export-declare-modifiers, @definitelytyped/no-single-element-tuple-type
+type Merge<T> = [{ [K in keyof T]: T[K] }][number];
+
+// eslint-disable-next-line @definitelytyped/strict-export-declare-modifiers
+type ExclusivePropertyUnion<T, P = keyof T> = P extends any
+    ? Merge<{ [K in Extract<keyof T, P>]: T[K] } & { [K in Exclude<keyof T, P>]?: never }>
+    : never;
+
 // http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_AttributeValue.html
-export interface AttributeValue {
-    B?: string | undefined;
-    BS?: string[] | undefined;
-    BOOL?: boolean | undefined;
-    L?: AttributeValue[] | undefined;
-    M?: { [id: string]: AttributeValue } | undefined;
-    N?: string | undefined;
-    NS?: string[] | undefined;
-    NULL?: boolean | undefined;
-    S?: string | undefined;
-    SS?: string[] | undefined;
-}
+export type AttributeValue = ExclusivePropertyUnion<{
+    B: Uint8Array;
+    BOOL: boolean;
+    BS: Uint8Array[];
```

Comment:
The [docs](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_AttributeValue.html) show this as an array of b64 encoded strings.